segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/ContainerRecoveryUtils.java

```diff
@@ -418,14 +419,13 @@ public static void updateCoreAttributes(Map<Integer, String> backUpMetadataSegme
      *                                  attributes segment has to be performed.
      * @param executorService           A thread pool for execution.
      * @param timeout                   Timeout for the operation.
-     * @return                          A Map of Container Ids to new container metadata segment names.
-     * @throws InterruptedException     If the operation was interrupted while waiting.
-     * @throws TimeoutException         If the timeout expired prior to being able to complete the operation.
-     * @throws ExecutionException       When execution of the opreations encountered an error.
+     * @return                          A CompletableFuture that, when completed normally, will have a Map of Container
```

Comment:
FYI: At line 404 above, you cannot return `null`; you need to return `CompletableFuture.<Void>completedFuture(null)`. Otherwise you'll get a (silent) NullPtrEx.